### PR TITLE
Cataclysm Mount DBCs

### PR DESCRIPTION
**Pull request by aaron02**
**Description**
added missing Mount DBCs within its required Spell cast cheks

**Todo / Checklist**
- [x] Target is branch develop.
- [x] Detailed description about this pull request.
- [x] Checked AE-Coding standards.

**Tests Performed:** 
- [x] Build AE.
- [x] Server startup.
- [x] Log into world.

### CODE REVIEW

File: src/world/GameCata/Storage/DBCStructures.h [hunk 1]
```diff
@@ -2105,6 +2111,26 @@ namespace DBC::Structures
         }
     };
 
+    struct MountCapabilityEntry
+    {
+        uint32_t  id;                                             // 0 index
```
Comment (schnek) [inline src/world/GameCata/Storage/DBCStructures.h:2116]: align //

File: src/world/GameCata/Storage/DBCStructures.h [hunk 2]
```diff
@@ -2105,6 +2111,26 @@ namespace DBC::Structures
...
+        uint32_t  reqAura;                                        // 4 required Aura
+        uint32_t  reqSpell;                                       // 5 spell that has to be known to you
+        uint32_t  speedModSpell;                                  // 6 spell to cast to apply mount speed effects
+        uint32_t  reqMap;                                         // 7 map where this is applicable
+    };
+
+    #define MAX_MOUNT_CAPABILITIES 24
+    struct MountTypeEntry
+    {
+        uint32  id;                                             // 0 index
```
Comment (schnek) [inline src/world/GameCata/Storage/DBCStructures.h:2129]: uint32_t  ?

File: src/world/GameCata/Storage/DBCStructures.h [hunk 3]
```diff
@@ -148,6 +148,12 @@ enum Targets
     TARGET_127                                  = 127,
 };
 
+enum MountFlags
+{
+    MOUNT_FLAG_CAN_PITCH    = 0x4,                    // client checks MOVEMENTFLAG2_FULL_SPEED_PITCHING
```
Comment (schnek) [inline src/world/GameCata/Storage/DBCStructures.h:153]: remove the spaces before //

File: src/world/Units/Unit.cpp [hunk 4]
```diff
@@ -6288,3 +6288,66 @@ void Unit::jumpTo(Object* obj, float speedZ, bool withOrientation)
...
+    uint32_t zoneId, areaId = 0;
+
+    zoneId = GetZoneId();
+    areaId = GetArea()->id;
+
+    uint32_t ridingSkill = 5000;
+    if (GetTypeFromGUID() == TYPEID_PLAYER)
+        ridingSkill = ToPlayer()->_GetSkillLineCurrent(SKILL_RIDING);
+
+    for (uint32 i = MAX_MOUNT_CAPABILITIES; i > 0; --i)
```
Comment (schnek) [inline src/world/Units/Unit.cpp:6311]: uint32_t ?

File: src/world/GameCata/Storage/DBCStores.cpp [hunk 5]
```diff
@@ -22,6 +22,9 @@ struct NameGenData
 
 std::vector<NameGenData> _namegenData[3];
 
+SERVER_DECL DBC::DBCStorage<DBC::Structures::MountCapabilityEntry>  sMountCapabilityStore(DBC::Structures::mount_capability_format);
```
Comment (schnek) [inline src/world/GameCata/Storage/DBCStores.cpp:25]: 1 space
image

**Review by schnek — APPROVED** [review]

File: src/world/Units/Unit.cpp [hunk 6]
```diff
@@ -6288,3 +6288,66 @@ void Unit::jumpTo(Object* obj, float speedZ, bool withOrientation)
     float speedXY = getExactDist2d(x, y) * 10.0f / speedZ;
     getMovementManager()->moveJump(x, y, z, getAbsoluteAngle(obj), speedXY, speedZ, EVENT_JUMP, withOrientation);
 }
+
+#if VERSION_STRING == Cata
+DBC::Structures::MountCapabilityEntry const* Unit::getMountCapability(uint32_t mountType)
+{
+    if (!mountType)
+        return NULL;
```
Comment (Zyres) [inline src/world/Units/Unit.cpp:6296]: Use nullptr, "NULL" is C-stylish

File: src/world/Units/Unit.cpp [hunk 7]
```diff
@@ -6288,3 +6288,66 @@ void Unit::jumpTo(Object* obj, float speedZ, bool withOrientation)
...
+
+    auto const* mountTypeEntry = sMountTypeStore.LookupEntry(mountType);
+    if (!mountTypeEntry)
+        return NULL;
```
Comment (Zyres) [inline src/world/Units/Unit.cpp:6300]: Use nullptr, "NULL" is C-stylish

File: src/world/Units/Unit.cpp [hunk 8]
```diff
@@ -6288,3 +6288,66 @@ void Unit::jumpTo(Object* obj, float speedZ, bool withOrientation)
...
+
+    uint32_t zoneId, areaId = 0;
```
Comment (Zyres) [inline src/world/Units/Unit.cpp:6302]: Why do we initialize these two to 0 when we override it the next lines anyway?

**Review by Zyres — CHANGES_REQUESTED** [review]

File: src/world/Units/Unit.cpp (hunk 8 above)
Comment (aaron02) [reply src/world/Units/Unit.cpp]: yeah was kinda useless you are true

**Review by Appled — APPROVED** [review]

**Pull request merged**